### ISSUE TITLE
RS028 Synchronized access to JSwingRipplesEIG node structure

### ISSUE THREAD

**Issue by javierobledo** [JSwingRipples2016/jswingripples#51]
This means make this object a synchronized resource, only accessed when it is ready. In the original JRipples plugin, this is achieved using boolean fields like "locked" or "ready" to determinate if the resource is used, ready or other kind of states.

**Comment by sgioia9** [comment]
You can use a mutex:

```
public class SomeClass {
private static final Object lock = new Object();

    xxx yyy someMethodWithACriticalSection() {
       ...
       synchronized(lock) {
           // crtitical section
           // access node structure here
        }
       ...
    }
}
```

**Comment by javierobledo** [comment]
no longer needed after #52

**Comment by javierobledo** [comment]
## Initialization:
Synchronized access to JSwingRipplesEIG node structure

**Comment by javierobledo** [comment]
## Concept Location

**Significant concepts:** Synchronized, JSwinRipplesEIG, Node

**Comment by javierobledo** [comment]
From JSwingRipplesEIG: History,JSwingRipplesEIGEdge, JSwingRipplesEIGEdgeEvent, JSwingRipplesEIGEvent, JSwingRipplesEIGListener, JSwingRipplesEIGNode, JSwingRipplesEIGNodeEvent, MemberHierarchySupportTest, NodeSearchBuilder, StartAnalysisDialog

**Comment by javierobledo** [comment]
## Prefactoring, Actualization, Postfactoring
No modification were made to the code

**Comment by javierobledo** [comment]
## Conclusion
This problem was fixed by another method, which was declared in the issue #50, with the pull request #52
